google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -44,7 +44,21 @@
 import java.util.Set;
 
 /**
- * Google Storage object metadata.
+ * Google Storage object information. A {@code BlobInfo} object includes the {@code BlobId} instance
+ * and the set of properties, such as the blob's access control configuration, user provided
+ * metadata, the CRC32C checksum, etc. Instances of this class are used to create a new Storage
+ * object or update the properties of an existing object. To deal with existing Storage objects the
+ * API includes the {@link Blob} class which extends {@code BlobInfo} and declares methods to
+ * perform operations on the object. Neither {@code BlobInfo} nor {@code Blob} instances keep the
+ * object content, just the object properties.
+ *
+ * <p>Example of usage {@code BlobInfo} to crate a Storage object:
+ *
+ * <pre>{@code
+ * BlobId blobId = BlobId.of(bucketName, blobName);
+ * BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/plain").build();
+ * Blob blob = storage.create(blobInfo, "Hello, world".getBytes(StandardCharsets.UTF_8));
```

Comment:
Can you please help me to understand this line: 

> Neither {@code BlobInfo} nor {@code Blob} instances keep the object content, just the object properties.

In this block, we are creating a string object of plain text type "Hello world". The `blob` object at line 60 refers to the byte[] of "Hello world". Do you think 'blob' keeping the said object here?

From what I conclude, the quoted statement and this example are contradicting.

Reply:
Thanks for looking!
"Hello world" - is the content of the new created blob, it's not stored withe the 'blob' object, it is located in the Storage. So, blob.getSize() will not cause RPC request, when blob.getContent() will.

Reply:
I will provide another example for the Blob class

Reply:
Thanks for clarifying my doubt!!